### PR TITLE
feat: improve elbow arrow keyboard move

### PR DESCRIPTION
**Pull request by zsviczian**
Fixes: Keyboard arrow keys stop to move elements when the group includes an elbow arrow #8387

Having looked at the code I concluded that this is not a bug, but a too restrictive feature. However, we can make the restriction more targeted/relaxed even without touching the arrow move logic. The keyboard move should only be restricted if the start or end element is not part of the selection.

### CODE REVIEW

File: packages/excalidraw/components/App.tsx
Comment (mtolmacs) [inline packages/excalidraw/components/App.tsx:4152]: `startMissing` is misleading here IMO, it's more like `startElementNotInSelection`.

File: packages/excalidraw/components/App.tsx
Comment (mtolmacs) [inline packages/excalidraw/components/App.tsx:4157]: Same here, it's misleading. I rather name it `endElementNotInSelection`.

File: packages/excalidraw/components/App.tsx
Comment (mtolmacs) [inline packages/excalidraw/components/App.tsx:4153]: Why check for this if in the `find()` predicate `el.id` will never equal `undefined` if the `startBinding` itself is `undefined`.

File: packages/excalidraw/components/App.tsx
Comment (mtolmacs) [inline packages/excalidraw/components/App.tsx:4158]: Why check for this if in the `find()` predicate `el.id` will never equal `undefined` if the `endBinding` itself is `undefined`.

**Review by mtolmacs — CHANGES_REQUESTED** [review]
This solves the specific issue this PR references, but don't we want to allow moving elbow arrows bound on only one end if the connected end boundable is also in the selection @dwelle ?

File: packages/excalidraw/components/App.tsx
Comment (zsviczian) [reply packages/excalidraw/components/App.tsx:4153]: my mistake, was intended to be .some((arrow) => ...

File: packages/excalidraw/components/App.tsx
Comment (zsviczian) [reply packages/excalidraw/components/App.tsx:4158]: and in that case find will return undefined, however for clarity I changed it to some.

File: packages/excalidraw/components/App.tsx
Comment (zsviczian) [reply packages/excalidraw/components/App.tsx]: changed it

File: packages/excalidraw/components/App.tsx
Comment (zsviczian) [reply packages/excalidraw/components/App.tsx]: changed it

File: packages/excalidraw/components/App.tsx
Comment (zsviczian) [reply packages/excalidraw/components/App.tsx]: ok. I am super slow today... I removed arrow.endBinding &&

Comment (mtolmacs) [reply packages/excalidraw/components/App.tsx]: Thanks. I'd wait for @dwelle whether he agrees with my feedback. I can see how pre-checking the existence of the binding can save on cycle count, but I'm on the fence whether it's worth it.

**Comment by zsviczian** [comment]
> This solves the specific issue this PR references, but don't we want to allow moving elbow arrows bound on only one end if the connected end boundable is also in the selection @dwelle ?

I agree - that would be ideal, however, this PR is already a big improvement.

File: packages/excalidraw/components/App.tsx
Comment (zsviczian) [reply packages/excalidraw/components/App.tsx:4158]: I told you I am slow today. `arrow.endBinding &&` is there because the element is only missing if there should be something bound, but it is not there.

**Comment by zsviczian** [comment]
> > This solves the specific issue this PR references, but don't we want to allow moving elbow arrows bound on only one end if the connected end boundable is also in the selection @dwelle ?
> 
> I agree - that would be ideal, however, this PR is already a big improvement.

I've modified the solution to include @mtolmacs's case as well. I think this is much better from a UX perspective.

**Comment by mtolmacs** [comment]
I can live with this, but I defer to @dwelle to give final approval. Thanks for the fix and flexibility!

**Comment by dwelle** [comment]
> I can live with this, but I defer to @dwelle to give final approval. Thanks for the fix and flexibility!

Haven't checked code, but from checking the preview it looks now arrow behavior is aligned to dragging behavior, which is what we want? Or is there something I'm missing?

**Comment by mtolmacs** [comment]
> > I can live with this, but I defer to @dwelle to give final approval. Thanks for the fix and flexibility!
> 
> Haven't checked code, but from checking the preview it looks now arrow behavior is aligned to dragging behavior, which is what we want? Or is there something I'm missing?

Yes, it makes sense to have the same behavior IMO.

**Comment by zsviczian** [comment]
Just to confirm, the current code in the PR fully aligns keyboard move with the dragging behavior for elbow arrows.

**Pull request merged**
